aptos-indexer-processors-sdk/testing-framework/src/mock_grpc.rs

```diff
@@ -0,0 +1,62 @@
+use aptos_protos::indexer::v1::{
+    raw_data_server::{RawData, RawDataServer},
+    GetTransactionsRequest, TransactionsResponse,
+};
+use futures::Stream;
+use std::pin::Pin;
+use tonic::{Request, Response, Status};
+
+#[derive(Default)]
+pub struct MockGrpcServer {
+    pub transactions: Vec<TransactionsResponse>,
+    pub chain_id: u64,
+}
+
+type ResponseStream = Pin<Box<dyn Stream<Item = Result<TransactionsResponse, Status>> + Send>>;
+
+#[tonic::async_trait]
+impl RawData for MockGrpcServer {
+    type GetTransactionsStream = ResponseStream;
+
+    async fn get_transactions(
+        &self,
+        req: Request<GetTransactionsRequest>,
+    ) -> Result<Response<Self::GetTransactionsStream>, Status> {
+        let version = req.into_inner().starting_version.unwrap();
+        let transaction = self.transactions.iter().find(|t| {
+            t.transactions.iter().any(|tx| {
+                println!("Checking transaction version: {}", tx.version);
+                tx.version == version
+            })
+        });
```

Comment:
Should we just always return `self.transactions`? It's kind of weird to only return one of the transactions. We can also make sure that `self.transactions` is sorted by version here.